05/DB.test.js

```diff
@@ -1 +1,110 @@
-import DB from './DB';
\ No newline at end of file
+import DB from './DB';
+
+describe('DB', () => {
+  let db;
+
+  beforeEach(() => {
+    db = new DB();
+  });
+
+  afterEach(() => {
+    db = null;
+  });
+
+  describe('Insert method', () => {
+    it('Should insert data and resolve with the inserted data', async () => {
+      const data = { id: 1, name: 'John' };
+      const result = await db.insert(data);
+      expect(result).toEqual(data);
+    });
+
+    it('Should generate an ID and resolve with the inserted data if ID is not provided', async () => {
+      const data = { name: 'Jane' };
+      const result = await db.insert(data);
+      expect(result.id).toBeDefined();
+      expect(result.name).toBe('Jane');
+    });
+
+    it('Should reject with an error if data has a non-numeric ID', async () => {
+      const data = { id: 'invalid', name: 'John' };
+      await expect(db.insert(data)).rejects.toMatch('ID can be only number!');
+    });
+
+    it('Should reject with an error if data with the same ID already exist', async () => {
+      const data1 = { id: 1, name: 'Alice' };
+      const data2 = { id: 1, name: 'Bob' };
+      await db.insert(data1);
+      await expect(db.insert(data2)).rejects.toMatch("ID can't be duplicated!");
+    });
+  });
+
+  describe('Select method', () => {
+    it('Should resolve with the selected data if the ID exists', async () => {
+      const data = { id: 1, name: 'Charlie' };
+      await db.insert(data);
+      const result = await db.select(1);
+      expect(result).toEqual(data);
+    });
+  });
+
+  describe('Remove method', () => {
+    it('Should resolve with a success message if the item is removed', async () => {
+      const data = { id: 1, name: 'David' };
+      await db.insert(data);
+      const result = await db.remove(1);
+      expect(result).toBe('Item was remove!');
```

Comment:
Sprawdzałbym jeszcze czy faktycznie baza jest pusta (`const rows = await db.getRows(); => rows.length === 0`), bo znów sam komunikat to jedno, ale nie wiemy czy faktycznie ten element został usunięty - trzeba to sprawdzić i być bardzo nieufny :D